addon/components/o-s-s/chip.ts

```diff
@@ -0,0 +1,66 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+export type SkinType = 'default' | 'primary' | 'success' | 'danger';
+
+type SkinDefType = {
+  [key in SkinType]: string;
+};
+
+export const SkinDefinition: SkinDefType = {
+  default: 'default',
+  primary: 'primary',
+  success: 'success',
+  danger: 'danger'
+};
+
+const BASE_CLASS_CHIP: string = 'upf-chip';
+interface OSSChipArgs {
+  skin?: string;
+  label?: string;
+  onClick(crossClick: boolean): void;
+  disabled?: boolean;
+  ellipsisWidth?: number;
+}
+
+export default class OSSChip extends Component<OSSChipArgs> {
+  constructor(owner: unknown, args: OSSChipArgs) {
```

Comment:
You need to assert that the `label` is present too. We don't want Chips with just the cross in it (+ tests)